tests/security-validation.test.ts

```
@@ -0,0 +1,348 @@
+
+      controlCharAttacks.forEach(attack => {
+        const searchResult = validateSearchQuery(attack);
+        const topicResult = validateTopicName(attack);
+
+        // All control character attacks should be flagged as invalid
+        expect(searchResult.isValid).toBe(false);
+        expect(topicResult.isValid).toBe(false);
+
+        // And should not contain control chars in sanitized output
+        // eslint-disable-next-line no-control-regex
+        expect(searchResult.sanitized).not.toMatch(/[\x00-\x1f\x7f-\x9f]/);
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)